### DISCUSSION TITLE
How do I get a Docker dev->staging->prod pipeline set up?

### DISCUSSION THREAD

**Discussion by corysimmons** [hasura/graphql-engine#8773] [Question] [answered]
Do you all just manually run Hasura CLI Console alongside dev in a separate tab?

Why isn't this part of core Hasura? Or at least make the Hasura CLI play nice in a Docker env with Hasura GraphQL Engine.

I just want to set up `yarn dev` and it fire up a Docker GraphQL Engine with Hasura CLI Console for auto-tracked migrations + metadata.

I know about all the `socat` hacks but I couldn't get any of them actually working consistently.

**Comment by adas98012** [comment]
Here is the doc to help you deploy Hasura GraphQL engine using docker:
https://hasura.io/docs/latest/deployment/deployment-guides/docker/

**Reply by corysimmons** [reply]
This doesn't help. It's okay, we finally found render.com (for Docker + Next.js) and are just going to run dev from multiple terminal tabs. This was painful though.

I don't know if I'll ever reach for Hasura again. I reached out so many times and read/tried so many things...

If you don't have a `dev->testing->review apps->staging->production` pipeline that is mainstream after this many years, you never will.

No idea how people code review anything if they use Hasura Cloud (yes, it offers regression testing but I don't see how that's equivalent), and it seems like that's Hasura's path now.

**Comment by adas98012** [comment, marked as answer]
To address your concern about review apps, Hasura does provide a preview option:

https://hasura.io/docs/latest/deployment/preview-apps/#preview-apps-github-action

Please try it out and let us know if it allays your concerns. Looking forward to hear back from your end.

**Reply by corysimmons** [reply]
Interesting. Thank you. Do migrations work with Hasura Cloud?

We need a way to code review changes to the db schema.
